### PR TITLE
C4998: constructor deduplication execution with test helpers

### PR DESCRIPTION
**Pull request by bigbes**

### CODE REVIEW

File: ledger-core/virtual/execute/execute.go [hunk 1]
```diff
@@ -311,10 +311,6 @@ func (s *SMExecute) stepTakeLock(ctx smachine.ExecutionContext) smachine.StateUp
 	}
 
 	if ctx.Acquire(executionSemaphore).IsNotPassed() {
-		if s.isConstructor {
```
Comment (sexyhamster) [inline ledger-core/virtual/execute/execute.go:314]: why you've removed this? how could you not acquire semaphor on object contruction?

File: ledger-core/virtual/handlers/vdelegatedcallrequest.go [hunk 2]
```diff
@@ -114,9 +114,9 @@ func (s *SMVDelegatedCallRequest) stepProcessRequest(ctx smachine.ExecutionConte
 	)
 
 	action := func(state *object.SharedState) {
-		if state.GetState() != object.HasState {
-			panic(throw.IllegalState())
-		}
+		// if state.GetState() != object.HasState {
```
Comment (sexyhamster) [inline ledger-core/virtual/handlers/vdelegatedcallrequest.go:117]: remove this >_<

File: ledger-core/testutils/journalinglogger/iterator.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		}
+	}
+}
+
+func (it *JournalIterator) WaitStop(sm smachine.StateMachine, count int) bool {
+	var declarationType = reflect.TypeOf(sm)
+
+	return it.WaitEvent(func(event debuglogger.UpdateEvent) bool {
+		if reflect.TypeOf(event.SM) == declarationType {
+			if event.Update.UpdateType == "stop" {
```
Comment (sexyhamster) [inline ledger-core/testutils/journalinglogger/iterator.go:46]: and `event.Update.UpdateType == "error"`

File: ledger-core/virtual/execute/execute.go (hunk 1 above)
Comment (bigbes) [reply ledger-core/virtual/execute/execute.go:314]: Please re-read that part and apply it to deduplication+AwaitDelegate

File: ledger-core/virtual/handlers/vdelegatedcallrequest.go (hunk 2 above)
Comment (bigbes) [reply ledger-core/virtual/handlers/vdelegatedcallrequest.go:117]: sure, but that commit would be partially overriden by https://github.com/insolar/assured-ledger/pull/331

File: ledger-core/testutils/journalinglogger/iterator.go (hunk 3 above)
Comment (bigbes) [reply ledger-core/testutils/journalinglogger/iterator.go:46]: sure

File: ledger-core/virtual/integration/vdelegated_test.go [hunk 4]
```diff
@@ -11,6 +11,7 @@ import (
 
 	"github.com/gojuno/minimock/v3"
 	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/vdelegated_test.go:14]: wrong assert

Comment (bigbes) [reply ledger-core/virtual/integration/vdelegated_test.go:14]: fixed

File: ledger-core/testutils/runner/logicless/execution.go [hunk 5]
```diff
@@ -92,12 +92,12 @@ func (m *ExecutionMock) next(expectedTp ExecutionChunkType) (*ExecutionChunk, er
 		currentCheckPosition = m.pos
 	)
 
-	if len(m.checks) < currentCheckPosition {
-		return nil, throw.E("unexpected next step 1")
+	if len(m.checks) <= currentCheckPosition {
+		return nil, throw.E("unexpected next step")
 	}
 
 	if m.checks[currentCheckPosition].tp != expectedTp {
```
[the remark this answers is not in the document]
Comment (bigbes) [reply ledger-core/testutils/runner/logicless/execution.go:99]: fixed

**Pull request merged**